lib/driver.js

```diff
@@ -287,15 +291,18 @@ class AndroidUiautomator2Driver extends BaseDriver {
 
   async startUiAutomator2Session () {
     // get device udid for this session
-    let {udid, emPort} = await helpers.getDeviceInfoFromCaps(this.opts);
+    const {udid, emPort} = await helpers.getDeviceInfoFromCaps(this.opts);
     this.opts.udid = udid;
     this.opts.emPort = emPort;
 
     // now that we know our java version and device info, we can create our
     // ADB instance
     this.adb = await androidHelpers.createADB(this.opts);
 
-    const apiLevel = await this.adb.getApiLevel();
+    // the initial access to adb may fail for the device being offline
+    const apiLevel = await retryInterval(ADB_INITIAL_RETRIES, 100, async () => {
```

Comment:
should we only do the retries if the env.CI variable is set?

Reply:
I don't think we should be checking that inside actual code (as opposed to our tests).

This does not add any overhead, really, if it returns the first time.

Reply:
this looks like a workaround to me. I would rather consider adding an option to `createADB` helper. This will address the issue for all android driver (since they all share the same helpers) and will make the stuff more flexible in general